Technomancy/tower.ts

```diff
@@ -392,12 +606,16 @@ function climb_tower(health = undefined, ehealth = undefined, grinding = false)
 
     }
 
-    if (adventure_data["tower_floor"] >= TOWER_DATA.length && !grinding) {
+    if (adventure_data["tower_floor"] >= TOWER_DATA.length - 2 + (tower_ascension_growth * adventure_data["tower_ascension"]) && !grinding) {
         $("#events_content").html("You're at the current top of the tower! Oh, also if you're here please message fuzzything44 on the Discord channel.<br/>");
-        
+        /* Reset the tower information, increment ascension count and reset cost of essence. */
+        adventure_data["tower_floor"]=1;
```

Comment:
This shouldn't be automatic. Give them an option to ascend (and say what it does, mainly that it makes the small tower harder). Also, ascension probably shouldn't reset total essence to 1 for balance reasons. 

Oh, also only show the button if there's rewards implemented for up to that ascension level probably.

Reply:
Can I just leave you to put the wrapping round this?   I can see two options for this.
1)  A confirm dialogue.
2)  A button to go through a 'portal' (or whatever) that does the reset.
There are probably other ways of doing it.

Reply:
Replying to myself! 
I've also added an attempt at making ascension optional.  Not sure whether it was the sort of thing you were thinking of.

Reply:
Yeah, you made it optional exactly how I would (a confirm dialogue would get annoying and could auto-ascend with the setting very easily, which isn't exactly what I want).

I'd still comment it out right now to completely disable ascension until rewards for ascending are implemented.

Reply:
O.k.  Can do. Although I thought that one of the rewards was extending the tower.  
Are there any rewards planned for the first ascension beyond resetting essence?  I.e. should I set it to be only possible for 0->1 ascension to start with?

Reply:
I'd say not possible to ascend at all until bosses are written and rewards are actually decided for first ascension. I definitely want to add AutoEssence buying at end of either Ascension I or II.